policy-controller/grpc/src/outbound.rs

```diff
@@ -372,86 +372,93 @@ fn to_proto(
 ) -> outbound::OutboundPolicy {
     let backend: outbound::Backend = default_backend(&policy, original_dst);
 
-    let kind = if policy.opaque {
-        outbound::proxy_protocol::Kind::Opaque(outbound::proxy_protocol::Opaque {
-            routes: vec![default_outbound_opaq_route(backend, &policy.parent_info)],
-        })
-    } else {
-        let accrual = policy.accrual.map(|accrual| outbound::FailureAccrual {
-            kind: Some(match accrual {
-                linkerd_policy_controller_core::outbound::FailureAccrual::Consecutive {
-                    max_failures,
-                    backoff,
-                } => outbound::failure_accrual::Kind::ConsecutiveFailures(
-                    outbound::failure_accrual::ConsecutiveFailures {
+    let kind = match &policy.app_protocol {
+        Some(AppProtocol::Opaque) => {
+            outbound::proxy_protocol::Kind::Opaque(outbound::proxy_protocol::Opaque {
+                routes: vec![default_outbound_opaq_route(backend, &policy.parent_info)],
+            })
+        }
+        None | Some(AppProtocol::Unknown(_)) => {
+            if let Some(AppProtocol::Unknown(protocol)) = &policy.app_protocol {
+                tracing::warn!(resource = ?policy.parent_info, port = policy.port.get(), "Unknown appProtocol \"{protocol}\"");
+            }
+
```

Comment:
```suggestion
            if let Some(AppProtocol::Unknown(protocol)) = &policy.app_protocol {
                tracing::debug!(resource = ?policy.parent_info, port = policy.port.get(), %protocol, "Unknown appProtocol");
            }

```

This isn't something that should pollute logs--it's okay and expected for applications to specify protocols that linkerd doesn't know about. At the end state of this work, after we add support for all known protocols, we'll want the unknown to case to be handled as Opaque. Until then, though, we'll temporarily continue to perform detection on these ports.

Reply:
Good call, though I'm still thinking we need some way of surfacing things like typos and accidental misconfiguration to the user.